### PR TITLE
[SymbolGraph] Print non-underscored attributes in declarations

### PR DESCRIPTION
**Pull request by bitjammer**
rdar://63338507

### CODE REVIEW

**Comment by bitjammer** [comment]
@swift-ci Please smoke test

**Comment by bitjammer** [comment]
@swift-ci Please smoke test

File: lib/SymbolGraphGen/SymbolGraph.cpp [hunk 1]
```diff
@@ -43,6 +43,7 @@ SymbolGraph::SymbolGraph(SymbolGraphASTWalker &Walker,
 
 // MARK: - Utilities
 
+
```
Comment (franklinsch) [inline lib/SymbolGraphGen/SymbolGraph.cpp:46]:
```suggestion
```

File: lib/SymbolGraphGen/SymbolGraph.cpp [hunk 2]
```diff
@@ -64,10 +65,35 @@ PrintOptions SymbolGraph::getDeclarationFragmentsPrintOptions() const {
...
+    ExcludeAttrs.insert(std::make_pair("DAK_" #CLASS, DAK_##CLASS));
+#define TYPE_ATTR(X) ExcludeAttrs.insert(std::make_pair("TAK_" #X, TAK_##X));
 #include "swift/AST/Attr.def"
 
+  // Allow the following type attributes:
+  ExcludeAttrs.erase("TAK_autoclosure");
+  ExcludeAttrs.erase("TAK_convention");
+  ExcludeAttrs.erase("TAK_noescape");
+  ExcludeAttrs.erase("TAK_escaping");
+  ExcludeAttrs.erase("TAK_inout");
```
Comment (franklinsch) [inline lib/SymbolGraphGen/SymbolGraph.cpp:81]: Where are these inserted in `ExcludeAttrs`?

Also, what do we do with other attributes, like `@objc`, `@main`, `@dynamicMemberLookup`, etc.?

File: test/SymbolGraph/Symbols/Mixins/DeclarationFragments/Subheading/Function.swift [hunk 3]
```diff
@@ -11,66 +11,66 @@ public func foo<S>(f: @escaping () -> (), ext int: Int = 2, s: S) where S: Seque
 // CHECK: names
 // CHECK-NEXT: "title": "foo(f:ext:s:)"
 // CHECK:      "subHeading": [
```
Comment (franklinsch) [inline test/SymbolGraph/Symbols/Mixins/DeclarationFragments/Subheading/Function.swift:13]: Should we also test attributes on functions; maybe by adding one to this function?

File: test/SymbolGraph/Symbols/Mixins/DeclarationFragments/Full/NominalTypes.swift [hunk 4]
```diff
@@ -6,11 +6,19 @@
 // RUN: %FileCheck %s --input-file %t/NominalTypes.symbols.json --check-prefix=ENUM
 // RUN: %FileCheck %s --input-file %t/NominalTypes.symbols.json --check-prefix=TYPEALIAS
 
-public struct S<T> where T: Sequence {}
+@frozen public struct S<T> where T: Sequence {}
```
Comment (franklinsch) [inline test/SymbolGraph/Symbols/Mixins/DeclarationFragments/Full/NominalTypes.swift:9]: Should we also test when a declaration has multiple attributes?

**Review by franklinsch — COMMENTED** [review]
Looks good, just a few comments

File: lib/SymbolGraphGen/SymbolGraph.cpp (hunk 2 above)
Comment (bitjammer) [reply lib/SymbolGraphGen/SymbolGraph.cpp:81]: These are not shown anywhere in the docs that I can see. The rest do occur from time to time, so they are not excluded here.

File: test/SymbolGraph/Symbols/Mixins/DeclarationFragments/Full/NominalTypes.swift (hunk 4 above)
Comment (bitjammer) [reply test/SymbolGraph/Symbols/Mixins/DeclarationFragments/Full/NominalTypes.swift:9]: It wouldn't hurt.

File: test/SymbolGraph/Symbols/Mixins/DeclarationFragments/Subheading/Function.swift (hunk 3 above)
Comment (bitjammer) [reply test/SymbolGraph/Symbols/Mixins/DeclarationFragments/Subheading/Function.swift:13]: I can add a test with an `@objc` attribute on a method, I think that would be a good case.

File: lib/SymbolGraphGen/SymbolGraph.cpp (hunk 2 above)
Comment (franklinsch) [reply lib/SymbolGraphGen/SymbolGraph.cpp:81]: `@propertyWrapper` is shown, for example: https://developer.apple.com/documentation/swiftui/state

I think others attributes might be shown, too.

Comment (bitjammer) [reply lib/SymbolGraphGen/SymbolGraph.cpp:81]: `@propertyWrapper` is a declaration attribute, not a type attribute. Declaration attributes are allowed by default here.

Comment (bitjammer) [reply lib/SymbolGraphGen/SymbolGraph.cpp:81]: (Except for underscored attributes of course)

**Review by franklinsch — APPROVED** [review]

**Comment by bitjammer** [comment]
@swift-ci Please smoke test

**Comment by bitjammer** [comment]
@swift-ci Please test Windows Platform

**Comment by bitjammer** [comment]
Pinged build czars about the Windows bot.

**Comment by bitjammer** [comment]
@swift-ci Please smoke test

**Comment by bitjammer** [comment]
@swift-ci Please smoke test OS X Platform

**Comment by bitjammer** [comment]
@swift-ci Please test Windows Platform

**Comment by bitjammer** [comment]
@swift-ci Please test Windows Platform

**Comment by bitjammer** [comment]
(Trying to land on a Windows bot other than 03)

**Comment by bitjammer** [comment]
@swift-ci Please test Windows Platform

**Pull request merged**